custom/inddex/food.py

```diff
@@ -195,7 +231,30 @@ def _set_composition(self):
 
         self.fct_reference_food_code_exists = bool(self.reference_food_code)
 
+    def set_fct_gap(self, ingredients=None):
+        if ingredients:
+            for row in ingredients:
+                row.set_fct_gap()
+
+        self.fct_gap_code = FctGaps.NOT_AVAILABLE
+
+        if self.food_type == FOOD_ITEM and self.fct_code:
+            self.fct_gap_code = {
+                'food_code': FctGaps.AVAILABLE,
+                'base_term_food_code': FctGaps.BASE_TERM,
+                'reference_food_code': FctGaps.REFERENCE,
+            }[self.fct_data_used]
+
+        if self.is_recipe:
+            if all(i.fct_gap_code == FctGaps.AVAILABLE for i in ingredients):
```

Comment:
unfortunately, calculating a gap code for a recipe requires context of the ingredients in that recipe, so I'm setting it during the `enrich` phase.